core/src/main/scala/org/apache/spark/util/random/RandomSampler.scala

```diff
@@ -41,6 +41,12 @@ trait RandomSampler[T, U] extends Pseudorandom with Cloneable with Serializable
   /** take a random sample */
   def sample(items: Iterator[T]): Iterator[U]
```

Comment:
I think it is doable. However, I am not sure we want to completely remove iterator-based implementation. @mengxr What do you think?